src/gitbackend.c

```diff
@@ -345,3 +345,70 @@ int init_repo(git_repository** repo, char* path, Archive* archive) {
     }
     return attach_archive_to_repo(*repo, archive);
 }
+
+#include <Python.h>
+
+typedef struct CRepository {
+    PyObject_HEAD
+    git_repository* repo;
+    PyObject *index;  /* It will be None for a bare repository */
+    PyObject *config; /* It will be None for a bare repository */
+    int owned;    /* _from_c() sometimes means we don't own the C pointer */
+} CRepository;
+
+
+int clone_object(PyObject* source_repo_object, PyObject* target_repo_object, const char* hex) {
+    git_odb* source_odb;
+    git_odb* target_odb;
+    git_repository* source_repo = ((CRepository*) source_repo_object)->repo;
+    git_repository* target_repo = ((CRepository*) target_repo_object)->repo;
+    int error = git_repository_odb(&source_odb, source_repo);
```

Comment:
yes but we do this once per project, this way I keep it nicely scopes and will get a clear python takctrack on the object that fialed if someting fails